### PR TITLE
[ACA-3134] - accept processs definition as input for start process

### PR DESCRIPTION
**Pull request by SilviuCPopa**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ACA-3134

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by eromano** [comment]
Not sure why we need it? @mauriziovitale @SilviuCPopa ?

**Comment by SilviuCPopa** [comment]
We need a possibility to pass the process definitions from others sources. Right now in ADW we call the `process-definitions` api every time we start a process.
In ADW would be nice to have a single api call and then pass the data from ngrx storage every time we start a process.
This is the PR in ADW https://github.com/Alfresco/alfresco-digital-workspace-app/pull/157

**Comment by eromano** [comment]
> We need a possibility to pass the process definitions from others sources. Right now in ADW we call the `process-definitions` api every time we start a process.
> In ADW would be nice to have a single api call and then pass the data from ngrx storage every time we start a process.
> This is the PR [Alfresco/alfresco-digital-workspace-app#157](https://github.com/Alfresco/alfresco-digital-workspace-app/pull/157)

This will move the service call outside the component I don't really like it . It's easy to break. It makes sense we don anyway this call the process can be updated and changed

**Comment by popovicsandras** [comment]
As I see it, passing down the definitions to this component, is not that problematic because, process definitions in AAE can be only changed if all of these steps happen:

1. A modeler persona changes a project in the modeling application
2. The same persona releases a new version of a project.
3. A devops persona is redeploying that particular application.

I think in the case of a new application version is deployed, we can expect the application user to refresh their application in the browser at least, since plenty of other things could have changed beside the process definitions. In ADW, we are making this request at least 3-4 times, getting back the same results. And these definitions are not that easy to be changed as I see it (above steps required).

@eromano , what do you think?

File: docs/process-services-cloud/components/start-process-cloud.component.md [hunk 1]
```diff
@@ -59,11 +60,11 @@ Starts a process.
 ### Starting a process with a default name and a pre-selected process definition name
 
 ```html
- <adf-cloud-start-process 
+ <adf-cloud-start-process
       [appId]="YOUR_APP_ID"
```
Comment (eromano) [inline docs/process-services-cloud/components/start-process-cloud.component.md:64]: appId doesn't exist is appName.  Is appname .
Can we add an example of this new property?

**Pull request closed without merging**